File: src/main/java/net/ndrei/teslacorelib/inventory/IMultiItemHandler.java
package net.ndrei.teslacorelib.inventory;

import net.minecraft.item.ItemStack;
import net.minecraftforge.items.IItemHandler;

/**
 * Created by CF on 2016-12-16.
 */
public interface IMultiItemHandler extends IFilteredItemHandler {
    int getInventories();
    IItemHandler getInventory(int inventory);
    IFilteredItemHandler getFilteredInventory(int inventory);

    boolean canInsertItem(int inventory, int slot, ItemStack stack);
    boolean canExtractItem(int inventory, int slot);
}

File: src/main/kotlin/net/ndrei/teslacorelib/inventory/IMultiItemHandler.kt
package net.ndrei.teslacorelib.inventory

import net.minecraft.item.ItemStack
import net.minecraftforge.items.IItemHandler

/**
 * Created by CF on 2017-06-28.
 */
interface IMultiItemHandler : IFilteredItemHandler {
    val inventories: Int
    fun getInventory(inventory: Int): IItemHandler
    fun getFilteredInventory(inventory: Int): IFilteredItemHandler?

    fun canInsertItem(inventory: Int, slot: Int, stack: ItemStack): Boolean
    fun canExtractItem(inventory: Int, slot: Int): Boolean
}
